notify/impl.go

```diff
@@ -682,6 +688,72 @@ func (n *OpsGenie) Notify(ctx context.Context, as ...*types.Alert) error {
 	return nil
 }
 
+// Pushover implements a Notifier for Pushover notifications.
+type Pushover struct {
+	conf *config.PushoverConfig
+	tmpl *template.Template
+}
+
+// NewPushover returns a new Pushover notifier.
+func NewPushover(c *config.PushoverConfig, t *template.Template) *Pushover {
+	return &Pushover{conf: c, tmpl: t}
+}
+
+func (*Pushover) name() string { return "pushover" }
+
+// Notify implements the Notifier interface.
+func (n *Pushover) Notify(ctx context.Context, as ...*types.Alert) error {
+	key, ok := GroupKey(ctx)
+	if !ok {
+		return fmt.Errorf("group key missing")
+	}
+	data := n.tmpl.Data(receiver(ctx), groupLabels(ctx), as...)
+
+	log.With("incident", key).Debugln("notifying Pushover")
+
+	var err error
+	tmpl := tmplText(n.tmpl, data, &err)
+
+	parameters := url.Values{}
+	parameters.Add("token", n.conf.Token)
+	parameters.Add("user", n.conf.UserKey)
+	title := tmpl(n.conf.Title)
+	message := tmpl(n.conf.Message)
+	parameters.Add("title", title)
+	if len(title)+len(message) > 512 {
```

Comment:
In this case the GeneratorURL is already covered in another field. It's unclear to me what we should do in cases where we've some space, but not enough to cover more than a handful of alerts. The common annotations and then start listing the alerts is best I guess. We may want to add some better logic for truncation than a simple cutoff, as there's quite a few fields across the various notifications methods that have limits.